covertToWordsNewVersion.js

```diff
@@ -0,0 +1,195 @@
+/*
+Create a function called convertToWords(number) it will receive a number and it will return a string  with the number written in letters.
+input: 1
+output: one
+First make it work with numbers from 0 to 10
+*/
+
+function convertToWords(number) {
+    let singleDigitsNumbersArray = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+    let singleDigitsNumbersNames = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
+
+    let TwoDigitsNumbersEnding = "teen";
+    let twentyAndPlus = "twenty";
+    let thirtyAndPlus = "thirty";
+    let fortyAndPlus = "forty";
+    let fiftyAndPlus = "fifty";
+    let sixtyAndPlus = "sixty";
+    let seventyAndPlus = "seventy";
+    let eightyAndPlus = "eighty";
+    let ninetyAndPlus = "ninety";
+    let oneHundredAndPlus = "one hundred"
+    let lastDigit = number - parseInt((number / 10)) * 10;
+
+    if (number < 10) {
+        for (let j = 0; j < singleDigitsNumbersArray.length; j++) {
+            if (number == singleDigitsNumbersArray[j]) {
+                return singleDigitsNumbersNames[j];
+            }
+        }
+    } else if (number >= 10 && number < 20) {
+        if (number == 10) {
+            return "ten";
+        } else {
+            for (let i = 0; i < singleDigitsNumbersArray.length; i++) {
+                if (lastDigit != 0) {
+                    if (lastDigit == 1) {
+                        return "eleven";
+                    } else {
+                        if (lastDigit == 2) {
+                            return "twelve";
+                        } else if (lastDigit == 3) {
+                            return "thir" + TwoDigitsNumbersEnding;
+                        } else if (lastDigit == 5) {
+                            return "fifth" + TwoDigitsNumbersEnding;
+                        } else {
+                            if (lastDigit == singleDigitsNumbersArray[i]) {
+                                return singleDigitsNumbersNames[i] + TwoDigitsNumbersEnding;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if (number >= 20 && number < 30) {
+        if (number == 20) {
+            return twentyAndPlus;
+        } else {
+            // let twentiesLastDigit = number - parseInt((number / 10)) * 10;
+            for (let k = 0; k < singleDigitsNumbersArray.length; k++) {
+                if (lastDigit != 0) {
+                    if (lastDigit == singleDigitsNumbersArray[k]) {
+                        return twentyAndPlus + "-" + singleDigitsNumbersNames[k];
+                    }
+                }
+            }
```

Comment:
Instead of duplicating the converting a single number into words we can have a function that converts a single number to words